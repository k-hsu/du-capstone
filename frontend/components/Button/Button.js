import React from "react";
import styled from "@emotion/styled";
import { spacing } from "../../theme";

const StyledButton = styled.button`
  background-color: white;
  border: 1px solid black;
  border-radius: ${spacing["0.25"]};
  padding: ${spacing["0.25"]} ${spacing["0.5"]};
`;

const Button = ({ children, ...props }) => {
  return <StyledButton {...props}>{children}</StyledButton>;
};

export default Button;
