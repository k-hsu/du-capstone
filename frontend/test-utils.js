import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
