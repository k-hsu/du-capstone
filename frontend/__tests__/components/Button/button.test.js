import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import Button from "../../../components/Button/Button";

describe("Button", () => {
  it("should render button component", () => {
    render(<Button>big red button</Button>);
    expect(screen.getByText("big red button")).toBeInTheDocument();
  });
  it("should render button component styles", () => {
    const component = renderer.create(<Button>big red button</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
