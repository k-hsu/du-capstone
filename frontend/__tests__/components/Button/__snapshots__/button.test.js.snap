// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button component styles 1`] = `
<button
  className="css-1f0mwd7"
>
  big red button
</button>
`;
